grpc/pb-marshalling.go

```diff
@@ -407,19 +407,20 @@ func registrationValid(reg *corepb.Registration) bool {
 }
 
 // orderValid checks that a corepb.Order is valid. In addition to the checks
-// from `newOrderValid` it ensures the order ID is not nil.
+// from `newOrderValid` it ensures the order ID and the BeganProcessing fields
+// are not nil.
 func orderValid(order *corepb.Order) bool {
-	return order.Id != nil && newOrderValid(order)
+	return order.Id != nil && order.BeganProcessing != nil && newOrderValid(order)
 }
 
 // newOrderValid checks that a corepb.Order is valid. It allows for a nil
 // `order.Id` because the order has not been assigned an ID yet when it is being
-// created initially. It also allows `order.CertificateSerial` to be nil such
-// that it can be used in places where the order has not been finalized yet.
-// Callers must additionally ensure the `CertificateSerial` field is non-nil if
-// they intend to use it.
+// created initially. It allows `order.BeganProcessing` to be nil because
```

Comment:
I'm not sure I follow this reasoning (I realize this is actually from #3402 but I spaced on it there), `sa.NewOrder` explicitly sets it to `false`, so why do we allow it to be `nil`?

Reply:
> `sa.NewOrder` explicitly sets it to `false`, so why do we allow it to be `nil`?

Because the RA calls `sa.NewOrder`, providing a `corepb.Order` as an argument. The SA's `NewOrder` RPC wrapper uses `newOrderValid` on the provided `corepb.Order` to decide if its well formed or not.

The choice is either:
A) Have the RA set `order.BeganProcessing` to false in the `corepb.Order` it uses for the `sa.NewOrder` rpc arg, and then update `newOrderValid` to reject `nil` `BeganProcessing`
B) Have the RA send a nil `order.BeganProcessing`, let `newOrderValid` allow that, and have the SA set `BeganProcessing` to a default.

I went with B. We could switch to A but it will require making the RA set the default. I don't have strong feelings either way. @rolandshoemaker Do you have a pref?

Reply:
Ah! I missed that, yeah I'm fine with the solution you've gone with here then.